node/lifecycle_test.go

```diff
@@ -103,13 +114,39 @@ func TestPodLifecycle(t *testing.T) {
 		}
 		t.Run("mockV0Provider", func(t *testing.T) {
 			assert.NilError(t, wireUpSystem(ctx, newMockV0Provider(), func(ctx context.Context, s *system) {
-				testCreateStartDeleteScenario(ctx, t, s)
+				testCreateStartDeleteScenario(ctx, t, s, waitForPodToBePermanentlyDeletedDeletionFunc)
 			}))
 		})
 	})
 
+	t.Run("createStartDeleteScenarioWithDeletionErrorNotFound", func(t *testing.T) {
+		// Wait for the pod to be started
+		mp := newMockProvider()
+		mp.errorOnDelete = errdefs.NotFound("not found")
+		assert.NilError(t, wireUpSystem(ctx, mp, func(ctx context.Context, s *system) {
+			testCreateStartDeleteScenario(ctx, t, s, waitForPodToBePermanentlyDeletedDeletionFunc)
+		}))
+	})
+
+	t.Run("createStartDeleteScenarioWithDeletionRandomError", func(t *testing.T) {
```

Comment:
It's not obvious to me what this test is supposed to do.  Retry of delete should succeed?

Reply:
It's supposed to ensure that the deletes are retried. I guess I could also check if the pod is (not) deleted in API server erroneously.

Reply:
The name of the test doesn't say, "testDeleteRetry" nor does any comment.